### ISSUE TITLE
Support for default values

### ISSUE THREAD

**Issue by ccamel** [goccy/go-yaml#61]
It appears that the library doesn't support default values. During unmarshaling, all the values not present in the yaml  are reset (or set to zero value).

`gopkg.in/yaml.v2` supports this behaviour, which is great.

See the following failing test:

```go
package yaml_test

import (
	"github.com/goccy/go-yaml"
	// "gopkg.in/yaml.v2"
	"strings"
	"testing"
)

func TestDecoder_DefaultValues(t *testing.T) {
	v  := struct {
		A string  `yaml:"a"`
		B string `yaml:"b"`
		c string // private
	}{
		B: "defaultBValue",
		c: "defaultCValue",
	}

	const src = `---
a: a_value
`
	if err := yaml.NewDecoder(strings.NewReader(src)).Decode(&v); err != nil {
		t.Fatalf(`parsing should succeed: %s`, err)
	}

	if v.A != "a_value" {
		t.Fatalf("v.A should be `a_value`, got `%s`", v.A)
	}

	if v.B != "defaultBValue" {
		t.Fatalf("v.B should be `defaultValue`, got `%s`", v.B)
	}

	if v.c != "defaultCValue" {
		t.Fatalf("v.c should be `defaultCValue`, got `%s`", v.c)
	}
}
```

Thanks.

**Comment by goccy** [comment]
@ccamel 

Thanks for the reporting !
I fixed with #62  and released as v1.1.5 .

**Comment by ccamel** [comment]
Great ! Thanks @goccy :+1:
